src/bp/core/module-loader.ts

```diff
@@ -100,24 +103,69 @@ export class ModuleLoader {
 
     for (const module of modules) {
       const name = _.get(module, 'definition.name', '').toLowerCase()
-      try {
-        ModuleLoader.processModuleEntryPoint(module, name)
-        const api = await createForModule(name)
-        await (module.onServerStarted && module.onServerStarted(api))
-        initedModules[name] = true
-        this.entryPoints.set(name, module)
-
-        const resourceLoader = new ModuleResourceLoader(this.logger, name, this.ghost)
-        await resourceLoader.importResources()
-      } catch (err) {
-        this.logger.attachError(err).error(`Error in module "${name}" onServerStarted`)
-      }
+      initedModules[name] = await this._loadModule(module, name)
     }
 
     this.callModulesOnReady(modules, initedModules) // Floating promise here is on purpose, we are doing this in background
     return Object.keys(initedModules)
   }
 
+  public async reloadModule(moduleLocation: string, moduleName: string) {
+    const resolver = new ModuleResolver(this.logger)
+    const absoluteLocation = await resolver.resolve(moduleLocation)
+
+    await this._unloadModule(absoluteLocation, moduleName)
+
+    const entryPoint = resolver.requireModule(absoluteLocation)
+    const isModuleLoaded = await this._loadModule(entryPoint, moduleName)
+
+    // Module loaded successfully, we will process its regular lifecycle
+    if (isModuleLoaded) {
```

Comment:
I imagine you use `this.callModulesOnReady` as i assume it does the same thing ?

Reply:
callModulesOnReady calls the serverReady on all modules, here I call it only for the specific module